src/bcrypt/__init__.py

```diff
@@ -76,3 +76,30 @@ def hashpw(password, salt):
         raise ValueError("Invalid salt")
 
     return _bcrypt.ffi.string(hashed)
+
+
+def kdf(password, salt, desired_key_bytes, rounds):
+    if isinstance(password, six.text_type) or isinstance(salt, six.text_type):
+        raise TypeError("Unicode-objects must be encoded before hashing")
+
+    if len(password) == 0 or len(salt) == 0:
+        raise ValueError("password and salt must not be empty")
+
+    if desired_key_bytes <= 0 or desired_key_bytes > 512:
+        raise ValueError("desired_key_bytes must be 1-512")
+
+    if rounds < 1:
+        raise ValueError("rounds must be 1 or more")
+
+    key = _bcrypt.ffi.new("uint8_t[]", desired_key_bytes)
+    res = _bcrypt.lib.bcrypt_pbkdf(
+        password, len(password), salt, len(salt), key, len(key), rounds
+    )
+    _bcrypt_assert(res == 0)
+
+    return _bcrypt.ffi.buffer(key, desired_key_bytes)[:]
```

Comment:
I'm happy to make this more explicit if you have a suggestion. This is typically how we do copies via cffi in `pyca/cryptography` (and is documented as such in the cffi docs: http://cffi.readthedocs.io/en/latest/ref.html#ffi-buffer-ffi-from-buffer). In those docs it also suggests `bytes()` though, so if you think that's a clearer way to show that the buffer is being copied I'll go ahead and change it!